cmd/query/app/apiv3/grpc_handler.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2021 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiv3
+
+import (
+	"context"
+
+	"github.com/gogo/protobuf/types"
+
+	"github.com/jaegertracing/jaeger/cmd/query/app/querysvc"
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/proto-gen/api_v3"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+)
+
+// Handler implements api_v3.QueryServiceServer
+type Handler struct {
+	QueryService *querysvc.QueryService
+}
+
+var _ api_v3.QueryServiceServer = (*Handler)(nil)
+
+// GetTrace implements api_v3.QueryServiceServer's GetTrace
+func (h *Handler) GetTrace(request *api_v3.GetTraceRequest, stream api_v3.QueryService_GetTraceServer) error {
+	traceID, err := model.TraceIDFromString(request.GetTraceId())
+	if err != nil {
+		return err
+	}
+
+	trace, err := h.QueryService.GetTrace(stream.Context(), traceID)
+	if err != nil {
+		return err
+	}
+	resourceSpans := jaegerSpansToOTLP(trace.GetSpans())
+	return stream.Send(&api_v3.SpansResponseChunk{
+		ResourceSpans: resourceSpans,
+	})
+}
+
+// FindTraces implements api_v3.QueryServiceServer's FindTraces
+func (h *Handler) FindTraces(request *api_v3.FindTracesRequest, stream api_v3.QueryService_FindTracesServer) error {
+	query := request.GetQuery()
+	queryParams := &spanstore.TraceQueryParameters{
+		ServiceName:   query.GetServiceName(),
+		OperationName: query.GetOperationName(),
+		Tags:          query.GetAttributes(),
+		NumTraces:     int(query.GetNumTraces()),
+	}
+	if query.GetStartTimeMin() != nil {
```

Comment:
Given `start_time_min`, `start_time_max`  are optional, it looks like they'll default to the zero value of `time.Time` of `TraceQueryParameters`. Will the queries to the backing span store be valid with these zero values?

Reply:
They are required see https://github.com/jaegertracing/jaeger-idl/pull/80.

Reply:
should we return an error if they are nil?

Reply:
+1 done